service/turn.rest/src/test/java/org/kaazing/gateway/service/turn/rest/TestConfigCredentialsGeneratorsIT.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2007-2016, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.service.turn.rest;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TestRule;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.gateway.util.feature.EarlyAccessFeatures;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+import javax.crypto.spec.SecretKeySpec;
+import java.io.FileInputStream;
+import java.security.KeyStore;
+
+import static java.nio.charset.Charset.forName;
+import static org.kaazing.test.util.ITUtil.createRuleChain;
+
+public class TestConfigCredentialsGeneratorsIT {
+    private static final String ACCEPT_URL = "http://localhost:8000/";
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    private final K3poRule robot = new K3poRule();
```

Comment:
That would allow you to reuse this Clas for other misconfigurations.  Also, because this doesn't use K3po, it would be alright to make this a unit test... though it doesn't matter really as it does test parts of gateway.server so arguably it is an integration test...